### ISSUE TITLE
No such file or directory

### ISSUE THREAD

**Issue by mrmachine** [ambv/bitrot#9]
Might be some kind of race condition when run over folders where files are actively and rapidly changing?

```
Traceback (most recent call last):
  File "/Users/tailee/.virtualenvs/bitrot/bin/bitrot", line 10, in <module>
    execfile(__file__)
  File "/Users/tailee/Projects/bitrot/bin/bitrot", line 30, in <module>
    run_from_command_line()
  File "/Users/tailee/Projects/bitrot/src/bitrot.py", line 323, in run_from_command_line
    chunk_size=args.chunk_size,
  File "/Users/tailee/Projects/bitrot/src/bitrot.py", line 138, in run
    st = os.stat(p)
OSError: [Errno 2] No such file or directory: './.dropbox/instance1/filecache.dbx-journal'
```

**Comment by yang** [comment]
Yeah, I've been seeing this as well - it's possible due to time between gathering all paths (via os.walk) and opening/hashing each file in the path.

A related issue, also due to on-going churn in the FS, is the converse - files appearing that weren't seen by os.walk.  E.g., files can remain invisible to bitrot just by being moved/renamed.

An ideal solution is to start monitoring for all FS changes from before os.walk is called, and then handle those changes in addition to all the paths gathered via os.walk (repeating this so long as there are more changes being introduced).  However, this is more (platform-dependent) work that I believe is less important due to most high-churn FS regions belonging to application areas like the Chrome web cache (which is where I frequently see this error) - churned files are unlikely to be of high concern to bitrot users.

Hence, I believe the practical desired fix is to not fail when this happens / suppress the exception.

**Comment by ambv** [comment]
Fixed in 0.7.0 as Yang is suggesting.
